mlflow/_doctor.py

```diff
@@ -2,8 +2,9 @@
 import platform
 
 import click
-import pkg_resources
+import importlib_metadata
```

Comment:
Does the standard importlib.metadata not provide the functionality needed here?
Are we trying to capture PyPA installed packages?

Reply:
looks like it does: https://docs.python.org/3.8/library/importlib.metadata.html

Reply:
I only brought it up because the core lib implementation is only compatible with importlib_metadata as of version 4 (IIRC). They're on version 6 now (I think there's a 2 year lag between merging the 3rd party lib's features into core python). It also has some more advanced features with determining dependencies for .whl files.

Reply:
Got it, I think we should use `importlib_metadata`. 1.4 is too old.

<img width="735" alt="image" src="https://github.com/mlflow/mlflow/assets/17039389/e5b124a5-8561-43da-a4b4-8c565a64e42c">

Reply:
SGTM!